libs/core/src/components/sage-checkbox/sage-checkbox.tsx

```diff
@@ -0,0 +1,106 @@
+import { Component, h, Prop, Host } from '@stencil/core';
+
+@Component({
+  tag: 'sage-checkbox',
+  styleUrl: 'sage-checkbox.scss',
+  shadow: true,
+})
+export class SageCheckbox {
+  /**
+   * Whether or not the checkbox is checked.
+   */
+  @Prop() checked: boolean;
+
+  /**
+   * Whether or not the checkbox is disabled.
+   */
+  @Prop() disabled: boolean;
+
+  /**
+   * Whether or not the checkbox is invalid.
+   */
+  @Prop() invalid: boolean;
+
+  /**
+   * String used for checkbox `id` attribute and label `for` attribute.
+   */
+  @Prop() checkboxId: string;
+
+  /**
+   * String used for label text next to checkbox
+   */
+  @Prop() label: string;
+
+  /**
+   * String used for message below checkbox
+   */
+  @Prop() message: string;
+
+  /**
+   * String used for checkbox `name` attribute.
+   */
+  @Prop() name: string;
+
+  /**
+   * Whether or not checkbox should display as indeterminate. Prop is for visual styling only.
+   * Only JavaScript can set the objects `indeterminate` property. See [MDN Web Docs](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/checkbox#indeterminate_state_checkboxes)
+   */
+  @Prop() indeterminate: boolean;
+
+  /**
+   * Whether or not the checkbox is required.
+   */
+  @Prop() required: boolean;
+
+  /**
+   * The value of the checkbox that is submitted with a form.
+   */
+  @Prop() value: string;
+
+  private classNames() {
+    let className = `sage-checkbox`;
+
+    if (this.invalid) {
+      const invalidClassName = 'sage-checkbox--invalid';
+      className += ' ' + invalidClassName;
+    }
+
+    if (this.indeterminate) {
+      const indeterminateClassName = 'sage-checkbox--indeterminate';
+      className += ' ' + indeterminateClassName;
+    }
+
+    if (this.disabled) {
+      const disabledClassName = 'sage-checkbox--disabled';
+      className += ' ' + disabledClassName;
+    }
```

Comment:
Similar question to what I asked in the scss file - do we need modifier classes to target these states? If we do want to do it with classes should we explore using state classes? 

e.g. `is-checked`, `is-invalid`, etc